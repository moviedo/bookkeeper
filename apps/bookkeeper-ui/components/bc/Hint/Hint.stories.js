export default {
  title: "base components / bc-hint",
};

const Template = (args, { argTypes }) => {
  return {
    props: Object.keys(argTypes),
    template: `<bc-hint v-bind="$props">${args.text ?? ""}</bc-hint>`,
  };
};

export const Default = () => ({
  template: `
  <div>
    Element will be EMPTY by DEFAULT
    <bc-hint></bc-hint>
  </div>
  `,
});

export const Error = Template.bind({});
Error.args = {
  type: "error",
  text: "Some hint to user",
};

export const Warning = Template.bind({});
Warning.args = {
  type: "warning",
  text: "Some hint to user",
};
