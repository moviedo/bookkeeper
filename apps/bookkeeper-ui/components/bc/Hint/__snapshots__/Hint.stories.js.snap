// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base components / bc-hint Default smoke-test 1`] = `
<div>
  Element will be EMPTY by DEFAULT
  <p data-v-0195b81e
     class="text-sm font-normal error"
  >
    <svg data-v-0195b81e
         xmlns="http://www.w3.org/2000/svg"
         viewbox="0 0 20 20"
         fill="currentColor"
         aria-hidden="true"
         class="inline h-5 w-5 pr-1"
         style="display: none;"
    >
      <path data-v-0195b81e
            fill-rule="evenodd"
            d="M8.257 3.099c.765-1.36 2.722-1.36 3.486 0l5.58 9.92c.75 1.334-.213 2.98-1.742 2.98H4.42c-1.53 0-2.493-1.646-1.743-2.98l5.58-9.92zM11 13a1 1 0 11-2 0 1 1 0 012 0zm-1-8a1 1 0 00-1 1v3a1 1 0 002 0V6a1 1 0 00-1-1z"
            clip-rule="evenodd"
      >
      </path>
    </svg>
    <span data-v-0195b81e>
    </span>
  </p>
</div>
`;

exports[`base components / bc-hint Default smoke-test 2`] = `
<div>
  Element will be EMPTY by DEFAULT
  <p data-v-0195b81e
     class="text-sm font-normal error"
  >
    <svg data-v-0195b81e
         xmlns="http://www.w3.org/2000/svg"
         viewbox="0 0 20 20"
         fill="currentColor"
         aria-hidden="true"
         class="inline h-5 w-5 pr-1"
         style="display: none;"
    >
      <path data-v-0195b81e
            fill-rule="evenodd"
            d="M8.257 3.099c.765-1.36 2.722-1.36 3.486 0l5.58 9.92c.75 1.334-.213 2.98-1.742 2.98H4.42c-1.53 0-2.493-1.646-1.743-2.98l5.58-9.92zM11 13a1 1 0 11-2 0 1 1 0 012 0zm-1-8a1 1 0 00-1 1v3a1 1 0 002 0V6a1 1 0 00-1-1z"
            clip-rule="evenodd"
      >
      </path>
    </svg>
    <span data-v-0195b81e>
    </span>
  </p>
</div>
`;

exports[`base components / bc-hint Default smoke-test: base-components-bc-hint--default 1`] = `
<div>
  Element will be EMPTY by DEFAULT
  <p data-v-0195b81e
     class="text-sm font-normal error"
  >
    <svg data-v-0195b81e
         xmlns="http://www.w3.org/2000/svg"
         viewbox="0 0 20 20"
         fill="currentColor"
         aria-hidden="true"
         class="inline h-5 w-5 pr-1"
         style="display: none;"
    >
      <path data-v-0195b81e
            fill-rule="evenodd"
            d="M8.257 3.099c.765-1.36 2.722-1.36 3.486 0l5.58 9.92c.75 1.334-.213 2.98-1.742 2.98H4.42c-1.53 0-2.493-1.646-1.743-2.98l5.58-9.92zM11 13a1 1 0 11-2 0 1 1 0 012 0zm-1-8a1 1 0 00-1 1v3a1 1 0 002 0V6a1 1 0 00-1-1z"
            clip-rule="evenodd"
      >
      </path>
    </svg>
    <span data-v-0195b81e>
    </span>
  </p>
</div>
`;

exports[`base components / bc-hint Error smoke-test 1`] = `
<p data-v-0195b81e
   class="text-sm font-normal error"
   text="Some hint to user"
>
  <svg data-v-0195b81e
       xmlns="http://www.w3.org/2000/svg"
       viewbox="0 0 20 20"
       fill="currentColor"
       aria-hidden="true"
       class="inline h-5 w-5 pr-1"
       style
  >
    <path data-v-0195b81e
          fill-rule="evenodd"
          d="M8.257 3.099c.765-1.36 2.722-1.36 3.486 0l5.58 9.92c.75 1.334-.213 2.98-1.742 2.98H4.42c-1.53 0-2.493-1.646-1.743-2.98l5.58-9.92zM11 13a1 1 0 11-2 0 1 1 0 012 0zm-1-8a1 1 0 00-1 1v3a1 1 0 002 0V6a1 1 0 00-1-1z"
          clip-rule="evenodd"
    >
    </path>
  </svg>
  <span data-v-0195b81e>
    Some hint to user
  </span>
</p>
`;

exports[`base components / bc-hint Error smoke-test 2`] = `
<p data-v-0195b81e
   class="text-sm font-normal error"
   text="Some hint to user"
>
  <svg data-v-0195b81e
       xmlns="http://www.w3.org/2000/svg"
       viewbox="0 0 20 20"
       fill="currentColor"
       aria-hidden="true"
       class="inline h-5 w-5 pr-1"
       style
  >
    <path data-v-0195b81e
          fill-rule="evenodd"
          d="M8.257 3.099c.765-1.36 2.722-1.36 3.486 0l5.58 9.92c.75 1.334-.213 2.98-1.742 2.98H4.42c-1.53 0-2.493-1.646-1.743-2.98l5.58-9.92zM11 13a1 1 0 11-2 0 1 1 0 012 0zm-1-8a1 1 0 00-1 1v3a1 1 0 002 0V6a1 1 0 00-1-1z"
          clip-rule="evenodd"
    >
    </path>
  </svg>
  <span data-v-0195b81e>
    Some hint to user
  </span>
</p>
`;

exports[`base components / bc-hint Error smoke-test: base-components-bc-hint--error 1`] = `
<p data-v-0195b81e
   class="text-sm font-normal error"
   text="Some hint to user"
>
  <svg data-v-0195b81e
       xmlns="http://www.w3.org/2000/svg"
       viewbox="0 0 20 20"
       fill="currentColor"
       aria-hidden="true"
       class="inline h-5 w-5 pr-1"
       style
  >
    <path data-v-0195b81e
          fill-rule="evenodd"
          d="M8.257 3.099c.765-1.36 2.722-1.36 3.486 0l5.58 9.92c.75 1.334-.213 2.98-1.742 2.98H4.42c-1.53 0-2.493-1.646-1.743-2.98l5.58-9.92zM11 13a1 1 0 11-2 0 1 1 0 012 0zm-1-8a1 1 0 00-1 1v3a1 1 0 002 0V6a1 1 0 00-1-1z"
          clip-rule="evenodd"
    >
    </path>
  </svg>
  <span data-v-0195b81e>
    Some hint to user
  </span>
</p>
`;

exports[`base components / bc-hint Warning smoke-test 1`] = `
<p data-v-0195b81e
   class="text-sm font-normal warning"
   text="Some hint to user"
>
  <svg data-v-0195b81e
       xmlns="http://www.w3.org/2000/svg"
       viewbox="0 0 20 20"
       fill="currentColor"
       aria-hidden="true"
       class="inline h-5 w-5 pr-1"
       style
  >
    <path data-v-0195b81e
          fill-rule="evenodd"
          d="M8.257 3.099c.765-1.36 2.722-1.36 3.486 0l5.58 9.92c.75 1.334-.213 2.98-1.742 2.98H4.42c-1.53 0-2.493-1.646-1.743-2.98l5.58-9.92zM11 13a1 1 0 11-2 0 1 1 0 012 0zm-1-8a1 1 0 00-1 1v3a1 1 0 002 0V6a1 1 0 00-1-1z"
          clip-rule="evenodd"
    >
    </path>
  </svg>
  <span data-v-0195b81e>
    Some hint to user
  </span>
</p>
`;

exports[`base components / bc-hint Warning smoke-test 2`] = `
<p data-v-0195b81e
   class="text-sm font-normal warning"
   text="Some hint to user"
>
  <svg data-v-0195b81e
       xmlns="http://www.w3.org/2000/svg"
       viewbox="0 0 20 20"
       fill="currentColor"
       aria-hidden="true"
       class="inline h-5 w-5 pr-1"
       style
  >
    <path data-v-0195b81e
          fill-rule="evenodd"
          d="M8.257 3.099c.765-1.36 2.722-1.36 3.486 0l5.58 9.92c.75 1.334-.213 2.98-1.742 2.98H4.42c-1.53 0-2.493-1.646-1.743-2.98l5.58-9.92zM11 13a1 1 0 11-2 0 1 1 0 012 0zm-1-8a1 1 0 00-1 1v3a1 1 0 002 0V6a1 1 0 00-1-1z"
          clip-rule="evenodd"
    >
    </path>
  </svg>
  <span data-v-0195b81e>
    Some hint to user
  </span>
</p>
`;

exports[`base components / bc-hint Warning smoke-test: base-components-bc-hint--warning 1`] = `
<p data-v-0195b81e
   class="text-sm font-normal warning"
   text="Some hint to user"
>
  <svg data-v-0195b81e
       xmlns="http://www.w3.org/2000/svg"
       viewbox="0 0 20 20"
       fill="currentColor"
       aria-hidden="true"
       class="inline h-5 w-5 pr-1"
       style
  >
    <path data-v-0195b81e
          fill-rule="evenodd"
          d="M8.257 3.099c.765-1.36 2.722-1.36 3.486 0l5.58 9.92c.75 1.334-.213 2.98-1.742 2.98H4.42c-1.53 0-2.493-1.646-1.743-2.98l5.58-9.92zM11 13a1 1 0 11-2 0 1 1 0 012 0zm-1-8a1 1 0 00-1 1v3a1 1 0 002 0V6a1 1 0 00-1-1z"
          clip-rule="evenodd"
    >
    </path>
  </svg>
  <span data-v-0195b81e>
    Some hint to user
  </span>
</p>
`;
